bentoml/server/bento_api_server.py

```diff
@@ -195,21 +194,54 @@ def bento_service_api_wrapper(api, service_name, service_version):
     summary_name = str(service_name) + "_" + str(api.name)
     request_metric_time = Summary(summary_name, summary_name + " request latency")
 
+    def log_image(request, request_id):
+        img_prefix = 'image/'
+        log_folder = config['logging'].get('base_log_folder')
+
+        if request.content_type.startswith(img_prefix):
+            filename = '{request_id}-{timestamp}.{ext}'.format(
+                request_id=request_id,
+                timestamp=int(time.time()),
+                ext=request.content_type[len(img_prefix) :],
+            )
+            with open(os.path.join(log_folder, filename), 'w') as f:
+                f.write(request.get_data())
+
+        for name in request.files:
+            file = request.files[name]
+            if file and file.filename:
+                orig_filename = secure_filename(file.filename)
+                filename = '{request_id}-{timestamp}-{orig_filename}'.format(
+                    request_id=request_id,
+                    timestamp=int(time.time()),
+                    orig_filename=orig_filename,
+                )
+                file.save(os.path.join(log_folder, filename))
+
     def wrapper():
         with request_metric_time.time():
             request_id = str(uuid.uuid4())
+            # Assume there is not a strong use case for idempotency check here.
+            # Will revise later if we find a case.
+
+            if not config['logging'].getboolean('disable_logging_image'):
+                log_image(request, request_id)
+
             response = api.handle_request(request)
-            if response.status_code == 200:
-                prediction_logger.info(
-                    {
-                        "uuid": request_id,
-                        "service_name": service_name,
-                        "service_version": service_version,
-                        "api": api.name,
-                        "request": _request_to_json(request),
-                        "response": response.response,
-                    }
-                )
+
+            request_log = {
+                "uuid": request_id,
+                "service_name": service_name,
+                "service_version": service_version,
+                "api": api.name,
+                "request": _request_to_json(request),
```

Comment:
in the case of image request, can we add the path to saved files here?

Reply:
good idea. will do.